src/Agda2Hs/Compile/Type.hs

```diff
@@ -161,6 +162,16 @@ compileDom x a
 
 compileTeleBinds :: Telescope -> C [Hs.TyVarBind ()]
 compileTeleBinds tel =
-  forM (map (hsName . unArg) $ filter keepArg $ teleArgNames tel) $ \x -> do
-    checkValidTyVar x
-    return $ Hs.UnkindedVar () x
+  forM
+    ( fmap (\(argName, domType) -> (hsName . unArg $ argName, unDom domType))
+    . filter (\(argName, _domType) -> keepArg argName) 
+    $ teleArgNames tel `zip` (flattenTel tel :: [Dom Type]) )
+    $ \(x, t) -> do
+      checkValidTyVarName x
+      case unEl t of
+        Sort (Type _) -> return $ Hs.UnkindedVar () x
+        _ -> genericDocError =<< do
+          text "Kind of bound argument not supported:" <+> 
+            parens (text (Hs.prettyPrint x) <> text " : " <> prettyTCM t)
+
+  
```

Comment:
Consider using https://github.com/agda/fix-whitespace to make sure your whitespace is uniform